crates/nu-cli/src/commands/sort_by.rs

```diff
@@ -65,6 +65,18 @@ fn sort_by(args: CommandArgs, registry: &CommandRegistry) -> Result<OutputStream
         let (SortByArgs { rest }, mut input) = args.process(&registry).await?;
         let mut vec = input.drain_vec().await;
 
+        for sort_arg in rest.iter() {
+            let mut match_test = get_data_by_key(&vec[0], sort_arg.borrow_spanned());
```

Comment:
It fails because you are accessing 0th element without checking whether `vec` is empty. Check if it is empty or use first() on vec. This simple testcase `"" | sort-by abc`  will panic with current version.

Reply:
👍 to what @k-brk said.

You can see that we do a `vec.is_empty` check just below the code you added, so you'll just want to move your code below that check 🙂